### DISCUSSION TITLE
amqpConnection.request error

### DISCUSSION THREAD

**Discussion by cesarpir** [golevelup/nestjs#508] [Q&A]
Hello, I'm trying to use this.amqpConnection.request to receive a payload (list) but the message does not arrive in rabbitMQ, an error appears like this:

error:  TypeError: (0 , crypto_1.randomUUID) is not a function

**Producer:**

```
@Get('findAll')
  async findAll(@Query('page') page?: number, @Query('size') size?: number): Promise<any[]> {
       const requestOptions: RequestOptions = {
      exchange: 'mercado-clientes',
      routingKey: 'clientes.v1.key-consultar-clientes',
      payload: { page, size},
      timeout: 10000,
    }
    try {
      const response: any[] = await this.amqpConnection.request(requestOptions);
      return response;
    } catch (error) {
      console.error("error: ", error);
    }
  }
```
**Consumer:**
```
  @RabbitRPC({
        exchange: 'mercado-clientes',
        routingKey: 'clientes.v1.key-consultar-clientes',
        queue: 'clientes.v1-consultar-clientes',
        errorBehavior: MessageHandlerErrorBehavior.NACK
    })
    async findAll(@Payload() payload: any): Promise<Cliente[]> {
        const { page, size } = payload;
        this.logger.log(`Microservice - FindAll Customer page: ${page} size: ${size}`);
        return await this.clientesService.findAll(page, size);
    }
```

Can you help me and guide me where I might be going wrong, I understood that to get a response a response would be just that!
Commenting that in another method that I use to create a new record everything is ok, in this way everything is correct:

```
@Post()
  @UsePipes(ValidationPipe)
  create(@Body() createCliente: CreateClienteDto) {
    this.logger.log("Create Cliente: ", JSON.stringify(createCliente));
    this.amqpConnection.publish('mercado-clientes', 'clientes.v1.key-create-clientes', createCliente);
  }
```
I really appreciate if you can help me. 

"@golevelup/nestjs-rabbitmq": "^3.3.0",
"@nestjs/core": "^9.0.0",

**Comment by WonderPanda** [comment]
What version of Node are you running on? If you're getting an error that randomUuid is not a function I suspect that your node version is out of date

**Reply by cesarpir** [reply]
Exactly, you are correct, it was the node version that was 12 using nvm now yes I am using v16.17.0.

Thank you very much for the help, it was attention speech on my part.

Sorry for mistakes in English I'm not fluent :)

Hugs Jesse Carter
